test/e2e/config/docker.yaml

```diff
@@ -86,6 +109,7 @@ providers:
 variables:
   # default variables for the e2e test; those values could be overridden via env variables, thus
   # allowing the same e2e config file to be re-used in different prow jobs e.g. each one with a K8s version permutation
+  INIT_WITH_BINARY: "https://github.com/kubernetes-sigs/cluster-api/releases/download/v0.3.16/clusterctl-{OS}-{ARCH}"
```

Comment:
clusterctl v1alpha3 allows only upgrades between the v1alpha3 series (v1alpha3 is not aware of v1alpha4 changes)
clusterctl v1alpha4 allows upgrades from v1alpha3 and also upgrades between the v1alpha4 series

Given that, in this test we are going to use clusterctl v1alpha4 to upgrade from a v1alpha3 cluster, but in order to do so first we need to use clusterctl v1alpha3 to create such cluster. This variable is defined to set which version to use to create a the cluster.

I also agree this could be improved by defining some sort or "version labels" like they exist in Kubernetes CI, but getting this into this PR will expand the scope too much.
I will open an issue to track the introduction of "version labels".

Reply:
Okay, makes sense to me. Maybe we should rename the the variable a bit. Maybe something like `CLUSTERCTL_BINARY` (which could be used for other tests) or `CLUSTERCTL_UPGRADE_INIT_BINARY` (hints stronger to our use case) would be easier to grasp.

Reply:
@fabriziopandini *push* In case you want to change it, but it's fine for me to keep as is.

Reply:
I have added on top of the variable to make this more explicit